crates/bitwarden-crypto/src/allocator.rs

```diff
@@ -0,0 +1,159 @@
+use core::slice;
+use std::alloc::{GlobalAlloc, Layout};
+
+use zeroize::Zeroize;
+
+/// Custom allocator that zeroizes memory before deallocating it
+///
+/// This is highly recommended to be enabled when using the Bitwarden crates to avoid sensitive data
+/// persisting in memory after it has been deallocated.
+///
+/// This allocator is a decorator around another allocator.
+pub struct ZeroizingAllocator<Alloc: GlobalAlloc>(pub Alloc);
+
+unsafe impl<T: GlobalAlloc> GlobalAlloc for ZeroizingAllocator<T> {
+    unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
+        self.0.alloc(layout)
+    }
+
+    unsafe fn dealloc(&self, ptr: *mut u8, layout: Layout) {
+        slice::from_raw_parts_mut(ptr, layout.size()).zeroize();
+
+        self.0.dealloc(ptr, layout);
+    }
+
+    unsafe fn alloc_zeroed(&self, layout: Layout) -> *mut u8 {
+        self.0.alloc_zeroed(layout)
+    }
+}
+
+/*
+#[cfg(test)]
+mod tests {
```

Comment:
These tests are disabled since they work inconsitently. Sometimes the memory is freed and immediately overwritten.

It's never the same value as what used to be there though.

Reply:
We can mark the test as ignored instead. That way it won't be run by default, but can be run with `cargo test --ignored`:
```rust
#[test]
#[ignore = "It produces inconsistent results on some platforms"]
fn test_fn() {
}
```